[All Hosts] (manifest) document the logic of requirements (#5627)
OfficeDev/office-js-docs-pr: docs/develop/understand-requirement-configuration.md

```json
"extensions": [
    {
        "requirements": {
            "scopes": [ "mail", "workbook" ]
        }
    }
    --- Other child properties of extensions here.
]
```

But the add-in's functionality in Outlook uses APIs in the **Mailbox 1.11** requirement set, while it's functionality in Excel uses APIs in the **ExcelApi 1.10** requirement set. It's natural to think that a `"capabilities"` object should be added to the requirement object like the following JSON.

```json
"extensions": [
    {
        "requirements": {
            "scopes": [ "mail", "workbook" ],
            "capabilities": [ 
                {
                    "name": "Mailbox",
                    "minVersion": "1.11"
                },
                {
                    "name": "ExcelApi",
                    "minVersion": "1.10"
                }
            ],
        }
    }
    --- Other child properties of extensions here.
]
```

But this configuration ensures that the add-in/feature won't be available on *any* version of Office because **Mailbox** is supported only in Outlook and **ExcelApi** is supported only in Excel. So, there is no Office application that supports both. 

> To achieve the two goals of making the add-in installable on (1) Outlook, but available only Outlook versions that support **Mailbox 1.11**, and (2) Excel, but available only Excel versions that support **ExcelApi 1.10**, the capability objects must be moved out of the `"extensions.requirements"` object into one or more of the child `"extensions.{FEATURE}.requirements"` objects, where {FEATURE} is a child of `"extensions"`, such as `"runtimes"` or `"ribbons"`, etc. 

Reviewer: ```suggestion
To achieve the two goals of making the add-in installable on (1) Outlook, but available only in Outlook versions that support **Mailbox 1.11**, and (2) Excel, but available only in Excel versions that support **ExcelApi 1.10**, the "capabilities" objects must be moved out of the `"extensions.requirements"` object into one or more of the child `"extensions.{FEATURE}.requirements"` objects, where {FEATURE} is a child of `"extensions"`, such as `"runtimes"` or `"ribbons"`.
```

For a concrete example, let's extend the scenario to specify that the add-in implements function commands following the guidance in [Create add-in commands with the unified manifest for Microsoft 365](create-addin-commands-unified-manifest.md). Specifically, the add-in has the following characteristics.

> - The add-in has custom ribbon buttons in both Outlook and Excel that trigger a function command in a commands.js file.

Reviewer: ```suggestion
- The add-in has custom ribbon buttons in both Outlook and Excel that trigger a function command in a **commands.js** file.
```

- To support the function command, the manifest has an `"extensions.runtimes.actions.actionId"` property whose value is "doSomething". 

> - The `Office.onReady` method in the commands.js file tests the [Office.context.host](/javascript/api/office/office.context#office-office-context-host-member) and branches depending on whether it is Outlook or Excel. If it is Outlook, it calls [Office.Action.associate](/javascript/api/office/office.actions#office-office-actions-associate-member(1)) to link "doSomething" to a function named `doSomethingInOutlook`. If the host is Excel, it calls `associate` to link "doSomething" to a function named `doSomethingInExcel`.

Reviewer: ```suggestion
- The `Office.onReady` method in the **commands.js** file tests the [Office.context.host](/javascript/api/office/office.context#office-office-context-host-member) and branches depending on whether it's Outlook or Excel. If it's Outlook, it calls [Office.Action.associate](/javascript/api/office/office.actions#office-office-actions-associate-member(1)) to link "doSomething" to a function named `doSomethingInOutlook`. If the host is Excel, it calls `associate` to link "doSomething" to a function named `doSomethingInExcel`.
```

- To implement the ribbon buttons, the manifest initially has a single ribbon object in a ["ribbons"](/microsoft-365/extensibility/schema/extension-ribbons-array) array. And this ribbon object has a control with an ["actionId"](/microsoft-365/extensibility/schema/extension-common-custom-group-controls-item#actionid) property set to "doSomething".

> It is this last characteristic that needs to be changed to achieve the two goals. The following is the strategy. 

Reviewer: ```suggestion
It's this last characteristic that needs to be changed to achieve the two goals. The following is the strategy. 
```

## Apply this guidance to the add-in only manifest

The logic of how Office processes requirement configuration in the add-in only manifest is almost the same as it does for the unified manifest, but there are some differences. 

> [!NOTE]
> Basic guidance for limiting add-ins/features by Office applications and by requirement set support with the add-in only manifest is in [Specify Office applications and API requirements with the add-in only manifest](specify-office-hosts-and-api-requirements.md). For limiting the add-in/feature by form factor, see [Limit by form factor with the add-in only manifest](#limit-by-form-factor-with-the-add-in-only-manifest).

### Limit by Office application in the add-in only manifest

Outlook add-ins and task pane add-ins for all non-Outlook add-ins each have there own manifest schemas, so an Outlook add-in cannot be combined with any other Office application in the same add-in. Content add-ins also have their own manifest schema and are supported only in Excel and PowerPoint. The remainder of this section is about limiting by application *within* the categories of task pane add-in and content add-in. 

> The equivalent of the `"scopes"` property is the [Hosts](/javascript/api/manifest/hosts) element. Just as there can be a `"requirements.scopes"` as either a direct child of `"extensions"` or a child of one the other child properties of `"extensions"`, so too there can be a `<Hosts>` element in the either root of the add-in only manifest or in a child `<VersionOverrides>`. 

Reviewer: ```suggestion
The equivalent of the `"scopes"` property is the [Hosts](/javascript/api/manifest/hosts) element. Just as there can be a `"requirements.scopes"` as either a direct child of `"extensions"` or a child of one the other child properties of `"extensions"`, so too there can be a `<Hosts>` element in either the root of the add-in only manifest or in a child `<VersionOverrides>`. 
```
